qCC/ccCommandLineCommands.h

```diff
@@ -1338,6 +1339,52 @@ struct CommandSFColorScale : public ccCommandLineInterface::Command
 	}
 };
 
+struct CommandSFConvertToRGB : public ccCommandLineInterface::Command
+{
+	CommandSFConvertToRGB() : ccCommandLineInterface::Command("SF convert to RGB", COMMAND_SF_CONVERT_TO_RGB) {}
+
+	virtual bool process(ccCommandLineInterface& cmd) override
+	{
+		cmd.print("[SF CONVERT TO RGB]");
+
+		if (cmd.arguments().empty())
+			return cmd.error(QString("Missing parameter: boolean (whether to mix with existing colors or not) after \"-%1\"").arg(COMMAND_SF_CONVERT_TO_RGB));
+
+		QString mixWithExistingColorsStr = cmd.arguments().takeFirst().toUpper();
+		bool mixWithExistingColors = false;
+		if (mixWithExistingColorsStr == "TRUE")
+		{
+			mixWithExistingColors = true;
+		}
+		else if (mixWithExistingColorsStr != "FALSE")
+		{
+			return cmd.error(QString("Invalid boolean value after \"-%1\". Got '%2' instead of TRUE or FALSE.").arg(COMMAND_SF_CONVERT_TO_RGB, mixWithExistingColorsStr));
+		}
+
+		if (cmd.clouds().empty())
+			return cmd.error(QString("No point cloud on which to compute SF gradient! (be sure to open one with \"-%1 [cloud filename]\" before \"-%2\")").arg(COMMAND_OPEN, COMMAND_SF_CONVERT_TO_RGB));
+
+		for (size_t i = 0; i < cmd.clouds().size(); ++i)
+		{
+			ccHObject* ent = cmd.clouds()[i].getEntity();
+			ccPointCloud* pc = cmd.clouds()[i].pc;
+			if (pc->getCurrentDisplayedScalarField())
```

Comment:
I'm not sure the 'currrent displayed scalar field' is set in the command line mode (as it has no meaning in this mode). If people use the 'SET_ACTIVE_SF' command before yours, then the ```setCurrentScalarField``` method will be called. It sets the _in_ and _out_ scalar field to the same value. Therefore you can use any of those, even though I think it would be more logical to use the _out_ one in your function (as you 'read' the values). In this case you'll have to use ```getCurrentOutScalarField``` method.